python/pyiceberg/expressions/base.py

```diff
@@ -867,3 +877,205 @@ def visit_unbound_predicate(self, predicate) -> BooleanExpression:
 
     def visit_bound_predicate(self, predicate) -> BooleanExpression:
         return predicate
+
+
+ROWS_MIGHT_MATCH = True
+ROWS_CANNOT_MATCH = False
+IN_PREDICATE_LIMIT = 200
+
+
+def _from_byte_buffer(field_type: IcebergType, val: bytes):
+    if not isinstance(field_type, PrimitiveType):
+        raise ValueError(f"Expected a PrimitiveType, got: {type(field_type)}")
+    return from_bytes(field_type, val)
+
+
+class _ManifestEvalVisitor(BoundBooleanExpressionVisitor[bool]):
+    partition_fields: list[PartitionFieldSummary]
+    partition_filter: BooleanExpression
+
+    def __init__(self, partition_struct_schema: Schema, partition_filter: UnboundPredicate, case_sensitive: bool = True):
+        bound_partition_filter = partition_filter.bind(partition_struct_schema, case_sensitive)
```

Comment:
This shouldn't call `bind` on the expression. It should use the `BindVisitor` to bind to the partition type.

I think it just wasn't obvious how to bind an expression because there isn't a helper method to do the binding and create a bind visitor. Probably just need to add that.

Reply:
Ah I see, I've updated the code 👍🏻